pkg/metaserver/agent/metric/provisioner/malachite/realtime_provisioner.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package malachite
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/kubewharf/katalyst-core/pkg/config/agent/global"
+	"github.com/kubewharf/katalyst-core/pkg/config/agent/metaserver"
+	"github.com/kubewharf/katalyst-core/pkg/consts"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver/agent/metric/provisioner/malachite/client"
+	malachitetypes "github.com/kubewharf/katalyst-core/pkg/metaserver/agent/metric/provisioner/malachite/types"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver/agent/metric/types"
+	"github.com/kubewharf/katalyst-core/pkg/metaserver/agent/pod"
+	"github.com/kubewharf/katalyst-core/pkg/metrics"
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+	utilmetric "github.com/kubewharf/katalyst-core/pkg/util/metric"
+)
+
+const (
+	metricsNamMalachiteRealtimeUnHealthy        = "malachite_realtime_unhealthy"
+	malachiteRealtimeProvisionerHealthCheckName = "malachite_realtime_provisioner_sample"
+	malachiteRealtimeProvisionTolerationTime    = 5 * time.Second
+	metricsNameMalachiteGetRealtimePowerFailed  = "malachite_get_realtime_power_failed"
+)
+
+func NewMalachiteRealtimeMetricsProvisioner(baseConf *global.BaseConfiguration, metricConf *metaserver.MetricConfiguration,
```

Comment:
Do we plan to pass the sampling interval from command line arguments?

Reply:
yes the arg should be like "--metric-provisioner-intervals=malachite_reltime=1" (this is to specify the interval to be 1 second instead of the default value)

Reply:
change code to set the default interval for malachite_realtime to 1 sec. so no need for explicit arg unless the 1 sec is not desired.